packages/playout-gateway/src/tsrHandler.ts

```diff
@@ -655,14 +657,31 @@ export class TSRHandler {
 				debugLoggingPs.push(device.setDebugLogging(debug))
 			}
 		}
+		// Set debugState on devices:
+		for (const device of this.tsr.getDevices()) {
+			const options: DeviceOptionsAny | undefined = deviceOptions.get(device.deviceId)
+			if (!options) {
+				continue
+			}
+
+			const debug: boolean = this.getDeviceDebugState(options)
+			if (device.debugState !== debug) {
+				this.logger.info(`Setting debugState of device ${device.deviceId} to ${debug}`)
+				debugLoggingPs.push(device.setDebugState(debug))
+			}
+		}
 		await Promise.all(debugLoggingPs)
+		await Promise.all(ps)
```

Comment:
We already await this on line 615 so it seems this line should not be necessary?

Reply:
Good spot, removed